src/python/grpcio_health_checking/grpc_health/v1/health.py

```diff
@@ -59,20 +60,35 @@ def close(self):
             self._condition.notify()
 
 
+def _watcher_to_on_next_adapter(watcher):
+
+    def on_next(response):
+        if response is None:
+            watcher.close()
+        else:
+            watcher.add(response)
+
+    return on_next
+
+
 class HealthServicer(_health_pb2_grpc.HealthServicer):
     """Servicer handling RPCs for service statuses."""
 
-    def __init__(self):
+    def __init__(self,
+                 experimental_non_blocking=True,
+                 experimental_thread_pool=None):
         self._lock = threading.RLock()
         self._server_status = {}
-        self._watchers = {}
+        self._on_next_callbacks = {}
+        self.Watch.__func__.experimental_non_blocking = experimental_non_blocking
+        self.Watch.__func__.experimental_thread_pool = experimental_thread_pool
```

Comment:
How are the decorators proposed by @gnossen going?

Reply:
They are not going, as far as I know. No point using decorators until/if this gets generalized to more use - a decorator for use in this one place will just make things more complicated, in my opinion.